libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -2560,7 +2560,7 @@ void IRGeneratorForStatements::appendExternalFunctionCall(
 
 	// Always use the actual return length, and not our calculated expected length, if returndatacopy is supported.
 	// This ensures it can catch badly formatted input from external calls.
-	if (m_context.evmVersion().supportsReturndata())
+	if (m_context.evmVersion().supportsReturndata() || returnInfo.dynamicReturnSize)
```

Comment:
I think it should be `&&`

Reply:
Oh actually no: If supportsReturndata and it is not dynamic, then we have to use something like `max(<returnSize>, returnsize())`

Reply:
Edit : Based on the discussion on the group, I've changed it to  min(<returnSize>, returndatasize())